modules/ROOT/pages/design-common-problems-raml-08-10.adoc

```diff
@@ -0,0 +1,350 @@
+= Common Problems in Conforming Either to RAML 0.8 or 1.0
+
+[[bookmark-a,Back to the top]]
+
+
+As of January 10, 2019, the API editor (one of the two editors in Design Center's API Designer) uses stricter validation for specifications that are written in RAML. If you open in the API editor a published API specification that is written in either RAML 0.8 or RAML 1.0 and you see error or warning indicators, this list might help you to resolve the problems in your RAML.
+
+
+// Review from here down
+
+== Declaring a URI parameter that is never used
+
+If an API specification declares a URI parameter, but then does not use that parameter, the editor displays this warning message:
+
+----
+unused uri parameter “parameter”
+----
+
+If the parameter is declared as a base URI parameter, but is not used, then this is the warning message:
+----
+unused base uri parameter “parameter”
+----
+
+For example, the following API specification would generate two warning messages:
+
+----
+unused uri parameter "unusedParam"
+----
+
+----
+unused base uri parameter "unusedUriParam"
+----
+
+----
+#%RAML 1.0
+title: test
+
+baseUri: http://param.raml/a/{baseUriParam1}/{nonExists}/{baseUriParam2}
+
+baseUriParameters:
+ baseUriParam1:
+   type: string
+ baseUriParam2:
+     type: string
+ unusedParam:
+   type: string
+
+/endpoint/{uriParam1}/{nonExistsUri}:
+ uriParameters:
+   uriParam1:
+     type: string
+   unusedUriParam:
+     type: string
+----
+
+To resolve the warning messages, you would simply need to remove the lines that declare these parameters:
+
+----
+#%RAML 1.0
+title: test
+
+baseUri: http://param.raml/a/{baseUriParam1}/{nonExists}/{baseUriParam2}
+
+baseUriParameters:
+ baseUriParam1:
+   type: string
+ baseUriParam2:
+     type: string
+
+/endpoint/{uriParam1}/{nonExistsUri}:
+ uriParameters:
+   uriParam1:
+     type: string
+----
+
+<<Back to the top>>
+
+== Not putting quotation marks around integers that are used as example values for strings
+
+If an example value is required to be a `string`, but the value is instead an `integer`, then the editor displays this error message:
+
+----
+expecting !!str and !!int provided
+----
+
+For example, the example value for `a` must be in quotation marks for the editor to parse it as a `string`:
+----
+#%RAML 1.0
```

Comment:
Change the example to this two:

#%RAML 1.0
title: test
types:
  400:
    type: string

To

#%RAML 1.0
title: test
types:
  '400':
    type: string



And



#%RAML 1.0
title: test
types:
  A:
    properties:
      a: string
      b: integer
    examples:
      1:
        a: some
        b: 16
      2:
        a: other
        b: 32

To

#%RAML 1.0
title: test
types:
  A:
    properties:
      a: string
      b: integer
    examples:
      '1':
        a: some
        b: 16
      '2':
        a: other
        b: 32

Reply:
I've made the changes for this comment. Please have a look.